lib/active_merchant/billing/gateways/credorax.rb

```diff
@@ -270,13 +270,38 @@ def add_email(post, options)
       def add_3d_secure(post, options)
         if options[:eci] && options[:xid]
           add_3d_secure_1_data(post, options)
+        elsif options[:execute_threed]
+          three_ds_2_options = options[:three_ds_2]
+          browser_info = three_ds_2_options[:browser_info]
```

Comment:
should we check that there is a :three_ds_2 key in the options hash before proceeding in this part to prevent errors, maybe in the elsif add ` && options[:three_ds_2]` or something similar

Reply:
Yeah shouldn't this be `options[:three_d_secure]` , and condition on `version`? https://github.com/activemerchant/active_merchant/wiki/Standardized-3DS-Fields

Reply:
@curiousepic We pass a different hash of values for 3DS direct.

Reply:
Step 5: https://docs.spreedly.com/guides/3dsecure2/